Tests/UnitTests/Misc/SigningTests.swift

```diff
@@ -58,4 +61,69 @@ class SigningTests: TestCase {
         })
     }
 
+    func testVerifySignatureWithInvalidSignatureReturnsFalseAndLogsError() throws {
+        let logger = TestLogHandler()
+
+        let message = "Hello World"
+        let nonce = "nonce"
+        let signature = "this is not a signature"
+
+        expect(Signing.verify(message: message.asData,
+                              nonce: nonce.asData,
+                              hasValidSignature: signature,
+                              with: try Signing.loadPublicKey())) == false
+
+        logger.verifyMessageWasLogged("Signature is not base64: \(signature)")
+    }
+
+    func testVerifySignatureWithInvalidSignature() throws {
+        expect(Signing.verify(message: "Hello World".asData,
+                              nonce: "nonce".asData,
+                              hasValidSignature: "invalid signature".asData.base64EncodedString(),
+                              with: try Signing.loadPublicKey())) == false
+    }
+
+    func testVerifySignatureLogsWarningWhenFail() throws {
+        let logger = TestLogHandler()
+
+        _ = Signing.verify(message: "Hello World".asData,
+                           nonce: "nonce".asData,
+                           hasValidSignature: "invalid signature".asData.base64EncodedString(),
+                           with: try Signing.loadPublicKey())
+
+        logger.verifyMessageWasLogged("Signature failed validation", level: .warn)
```

Comment:
I love that we have tests for logging stuff now

Reply:
💯